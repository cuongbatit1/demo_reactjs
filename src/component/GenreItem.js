import { memo } from "react";
import PropTypes from "prop-types";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import { CardActionArea } from "@mui/material";

const GenreItem = (props) => {
  const { text, index } = props;
  const random = index % 3;
  let linearGradient = "linear-gradient(135.46deg, #00CBCF 0.78%, #007AD9 100%)";
  if (random === 0) {
    linearGradient = "linear-gradient(132.93deg, #7591E8 12.67%, #87CEF8 107.3%)";
  } else if (random === 1) {
    linearGradient = "linear-gradient(132.93deg, #FF80B5 12.67%, #FF5A9F 107.3%)";
  }
  return (
    <>
      {console.log("GenreItem Render ", text)}
      <Card
        sx={{
          width: 140,
          height: 80,
          background: linearGradient,
        }}
      >
        <CardActionArea sx={{ height: "100%", p: 0 }}>
          <CardContent sx={{ p: 0 }}>
            <Typography
              sx={{ color: "white", m: 0 }}
              gutterBottom
              align="center"
              variant="h6"
              component="div"
            >
              {text}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </>
  );
};

GenreItem.propTypes = {
  text: PropTypes.string.isRequired,
};
export default memo(GenreItem);
