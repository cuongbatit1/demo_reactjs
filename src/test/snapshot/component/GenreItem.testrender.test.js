import { render, screen } from "@testing-library/react";
import GenreItem from "../../../component/GenreItem";
// random in component then hard testing snapshot
test("test renders GenreItem snapshot with index 0", () => {
  const { asFragment } = render(<GenreItem text="A" index={0} />);
  expect(asFragment()).toMatchSnapshot();
});

test("test renders GenreItem snapshot with index 1", () => {
  const { asFragment } = render(<GenreItem text="AA" index={1} />);
  expect(asFragment()).toMatchSnapshot();
});

test("test renders GenreItem snapshot with index 2", () => {
  const { asFragment } = render(<GenreItem text="AAA" index={2} />);
  expect(asFragment()).toMatchSnapshot();
});
