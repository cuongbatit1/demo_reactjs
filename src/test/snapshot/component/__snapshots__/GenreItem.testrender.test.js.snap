// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders GenreItem snapshot with index 0 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-ox2vj5-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-gpgb4u-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiTypography-gutterBottom css-8uu8ll-MuiTypography-root"
        >
          A
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`test renders GenreItem snapshot with index 1 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1r5tgcg-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-gpgb4u-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiTypography-gutterBottom css-8uu8ll-MuiTypography-root"
        >
          AA
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`test renders GenreItem snapshot with index 2 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-k63y40-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-gpgb4u-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root css-1i9wt8r-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiTypography-gutterBottom css-8uu8ll-MuiTypography-root"
        >
          AAA
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
